src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -0,0 +1,338 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
+{
+
+    /// <summary>
+    /// The exported CPS debugger for all types of K projects (web, consoles, classlibraries). Defers to
+    /// other types to get the DebugTarget information to launch.
+    /// </summary>
+    [ExportDebugger(ProjectDebugger.SchemaName)]
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    internal class ProjectDebuggerProvider : DebugLaunchProviderBase, IDeployedProjectItemMappingProvider
+    {
+
+        /// <summary>
+        /// Constructors. Unit test one is 2nd
+        /// </summary>
+        [ImportingConstructor]
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider)
+            : base(configuredProject)
+        {
+            LaunchSettingsProvider = launchSettingsProvider;
+
+            // We want it sorted so that higher numbers come first (is the default for these collections but explicitly expressed here)
+            ProfileLaunchTargetsProviders = new OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst, 
+                                                                                                                    configuredProject.UnconfiguredProject);
+        }
+
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider, 
```

Comment:
If this is a Unit Test constructor, it shouldn't be exposed to the world, just internal